great_expectations/util.py

```diff
@@ -689,6 +689,7 @@ def read_feather(  # noqa: PLR0913
         )
 
 
+@public_api
```

Comment:
I think the Dataset stuff is very old and possibly already deprecated code.
I don't think we want the `@public_api` tags here.
```suggestion
```